src/libraries/Native/Unix/System.Security.Cryptography.Native/configure.cmake

```diff
@@ -12,6 +12,11 @@ check_function_exists(
 	SSL_get0_alpn_selected
 	HAVE_OPENSSL_ALPN)
 
+check_function_exists(
```

Comment:
This means that we must compile on a machine that has a modern version of OpenSSL, correct? Otherwise the package we produce might have all of these code paths ifdefed away, even if the consuming app has a compatible version of OpenSSL?

Reply:
I'd probably look to @bartonjs for clarification on that to say for sure... The portable / non-portable build is a tad bit muddled in my head.

Reply:
If the shim is built portable we force all the HAVE_s to true, and ask via API_EXISTS.

If the shim is built direct then we expect that the compile and runtime are the same environment, e.g. it's a shim built specifically for Fedora as part of a distro-provided build of .NET.  Since Fedora knows how their OpenSSL build is built, the compile-time excision makes sense.